src/App.js

```diff
@@ -1,16 +1,23 @@
-import { Routes, Route } from "react-router-dom";
-import React from "react";
+import React, { useEffect } from "react";
+import { Routes, Route, useLocation, useSearchParams } from "react-router-dom";
 
 import SideNav from "./components/side-nav/SideNav";
 import Header from "./components/header/Header";
 import Home from "./pages/home/Home";
 import Search from "./pages/search/Search";
 import Login from "./pages/login/Login";
-import { useLocation } from "react-router-dom";
 
 function App() {
   const location = useLocation();
   const shouldShowHeaderAndSidenav = location.pathname !== "/login";
+  const [searchParams] = useSearchParams();
+
+  useEffect(() => {
```

Comment:
the 'useEffect' hook, which listens for changes, is a hook that executes a block of code when something changes or a function is triggers. Essentially, the useEffect' hook lets you perform side effect functional components  when the application mounts or when state or prop gets updated. 

[useEffect Hook](https://www.educative.io/answers/how-to-use-the-reactjs-useeffect-hook?utm_campaign=brand_educative&utm_source=google&utm_medium=ppc&utm_content=performance_max&eid=5082902844932096&utm_term=&utm_campaign=%5BNew%5D+Performance+Max&utm_source=adwords&utm_medium=ppc&hsa_acc=5451446008&hsa_cam=18511913007&hsa_grp=&hsa_ad=&hsa_src=x&hsa_tgt=&hsa_kw=&hsa_mt=&hsa_net=adwords&hsa_ver=3&gclid=CjwKCAjw5dqgBhBNEiwA7PryaMbZG3iycyR8y6lKztGr5Vge3Ed4BQM_Gq_OQqVoAIRIEbZkmKlOURoC1JEQAvD_BwE)